### PR TITLE
Added DocumentListStyle command

### PR DESCRIPTION
**Pull request by arkflpc**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (list): Adds `DocumentListStyleCommand`. See #11166.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-list/src/documentlist/utils/listwalker.js [hunk 1]
```diff
@@ -95,6 +121,13 @@ export default class ListWalker {
 		 */
 		this._sameItemType = !!options.sameItemType;
 
+		/**
+		 * Whether should return only blocks with the same `listStyle`, `listReversed` and `listStart` attributes.
+		 * @private
+		 * @type {Boolean}
+		 */
+		this._sameProperties = !!options.sameProperties;
+
```
Comment (niegowski) [inline packages/ckeditor5-list/src/documentlist/utils/listwalker.js:130]: Those properties are optional feature-related and extend the set of base list attributes (indent, type, item id). I think that there is no point in extending ListWalker with additional attributes, especially since you could stop iterating it as soon as you encounter the abort element from outside of the walker itself.

File: packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	/**
+	 * Executes the command.
+	 *
+	 * @param {Object} options
+	 * @param {String|null} [options.type] The type of the list style, e.g. `'disc'` or `'square'`. If `null` is specified, the default
+	 * style will be applied.
+	 * @protected
+	 */
+	execute( options = {} ) {
+		this._tryToConvertItemsToList( options );
```
Comment (niegowski) [inline packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js:57]: You should wrap both model changes (changing the list type and changing the list style) in a single change block, otherwise, it would generate 2 separate undo steps.

**Review by niegowski — CHANGES_REQUESTED** [review]
Partial review (tests not verified).

File: packages/ckeditor5-list/tests/documentlist/documentlistcommand.js [hunk 3]
```diff
@@ -30,7 +30,11 @@ describe( 'DocumentListCommand', () => {
 		model.schema.register( 'blockQuote', { inheritAllFrom: '$container' } );
 		model.schema.extend( '$container', { allowAttributes: [ 'listType', 'listIndent', 'listItemId' ] } );
 
-		stubUid();
+		stub = stubUid();
```
Comment (niegowski) [inline packages/ckeditor5-list/tests/documentlist/documentlistcommand.js:33]: Why is this change needed? Isn't `testUtils.createSinonSandbox();` handling the restore correctly?

File: packages/ckeditor5-list/tests/documentlistproperties/documentliststylecommand.js [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+import { modelList } from '../documentlist/_utils/utils';
+import stubUid from '../documentlist/_utils/uid';
+
+describe( 'DocumentListStyleCommand', () => {
+	let editor, model, bulletedListCommand, numberedListCommand, listStyleCommand, stub;
+
+	beforeEach( async () => {
+		const newEditor = await VirtualTestEditor.create( {
+			plugins: [ Paragraph, DocumentListPropertiesEditing ]
+		} );
```
Comment (niegowski) [inline packages/ckeditor5-list/tests/documentlistproperties/documentliststylecommand.js:19]: You should not use the full editor instance to test commands. We are usually instantiating only required classes to test commands to make sure that the command itself is generating a correct model state. Otherwise, post-fixers would fix the model and we would not know if the command itself is generating the correct output.

**Review by niegowski — CHANGES_REQUESTED** [review]

**Comment by niegowski** [comment]
I pushed some proposals of changes: [6a84586](https://github.com/ckeditor/ckeditor5/pull/11232/commits/6a8458623f0cc2f2257e175ca91974fb158b1f82).

File: packages/ckeditor5-list/tests/documentlist/documentlistcommand.js (hunk 3 above)
Comment (arkflpc) [reply packages/ckeditor5-list/tests/documentlist/documentlistcommand.js:49]: Well, it didn't in documentliststylecommand, because I forgot to use this sandbox there... fixing

File: packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	refresh() {
+		this.value = this._getValue();
+		this.isEnabled = this._checkEnabled();
+	}
+
+	/**
+	 * Executes the command.
+	 *
+	 * @param {Object} options
+	 * @param {String|null} [options.type] The type of the list style, e.g. `'disc'` or `'square'`. If `null` is specified, the default
```
[the remark this answers is not in the document]
Comment (arkflpc) [reply packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js:52]: It would break the UI. And if UI is updated, it wouldn't work with `ListStyleCommand`. And if `ListStyleCommand` is updated, that would be breaking change.

File: packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+		return numberedList.isEnabled || bulletedList.isEnabled;
+	}
+
+	/**
+	 * Check if the provided list style is valid. Also change the selection to a list if it's not set yet.
+	 *
+	 * @param {Object} options
+	 * @param {String|null} [options.type] The type of the list style. If `null` is specified, the function does nothing.
+	 * @private
```
Comment (niegowski) [inline packages/ckeditor5-list/src/documentlistproperties/documentliststylecommand.js:120]:
```suggestion
	 * @private
	 * @param {Object} options
	 * @param {String|null} [options.type] The type of the list style. If `null` is specified, the function does nothing.
```
